acoustid.py

```diff
@@ -348,6 +353,55 @@ def fingerprint_file(path, maxlength=MAX_AUDIO_LENGTH, force_fpcalc=False):
         return _fingerprint_file_fpcalc(path, maxlength)
 
 
+
+
+
+
+
+def compare_fingerprints(a, b) -> float:
+    """
+    compare two fingerprints locally
+    :param a: fingerprint of acoustid.fingerprint_file(filepath_a)
+    :param b: second fingerprint of acoustid.fingerprint_file(filepath_b)
+    :return:  similarity score [0,1]
+    """
+
+    if not have_chromaprint:
+        raise ModuleNotFoundError("function needs chromaprint")
+
+    # decompress fingerprints
+    a = chromaprint.decode_fingerprint(a)[0]
+    b = chromaprint.decode_fingerprint(b)[0]
+    a = list(map(int, a))
+    b = list(map(int, b))
+
+    ACOUSTID_MAX_BIT_ERROR = 2
+    ACOUSTID_MAX_ALIGN_OFFSET = 120
+
+    def popcount(x):
+        return bin(x).count('1')
+
+    def match_fingerprints(a, b):
```

Comment:
Is there any particular reason to make this an "inner" function here? It seems like it would work well just to have this code a the top level of the main `compare_fingerprints` function.

Or, if it's nice to have separate for testability or similar reasons, maybe we should make it a separate top-level function in this module—and document how to invoke it directly, if that ever makes sense to do?